src/Middleware.php

```diff
@@ -3,31 +3,138 @@
 namespace Inertia;
 
 use Closure;
-use Illuminate\Support\Facades\Response;
-use Symfony\Component\HttpFoundation\RedirectResponse as Redirect;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 
 class Middleware
 {
-    public function handle($request, Closure $next)
+    /**
+     * Determines the current asset version.
+     *
+     * @see https://inertiajs.com/asset-versioning
+     * @param  \Illuminate\Http\Request  $request
+     * @return string|null
+     */
+    public function version(Request $request)
     {
-        $response = $next($request);
+        // When we are running on Laravel Vapor, asset URLs are automatically updated
+        // when the application is being deployed. Because of this, we do not need
+        // to use any files for hashing, as we can simply use this URL instead.
+        if (config('app.asset_url')) {
+            return md5(config('app.asset_url'));
+        }
 
-        if (!$request->header('X-Inertia')) {
-            return $response;
+        // Alternatively, when we are running in a regular (non-serverless) environment
+        // we'll attempt to use the Laravel Mix asset manifest to generate our hash.
+        if (file_exists($manifest = public_path('mix-manifest.json'))) {
+            return md5_file($manifest);
         }
+    }
+
+    /**
+     * Defines the props that are shared by default.
+     *
+     * @see https://inertiajs.com/shared-data
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function share(Request $request)
+    {
+        return [
+            'errors' => function () use ($request) {
+                return $this->resolveValidationErrors($request);
+            },
+        ];
+    }
+
+    /**
+     * Handle the incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  Closure  $next
+     * @return Response
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        Inertia::share($this->share($request));
+
+        Inertia::version(function () use ($request) {
+            return $this->version($request);
+        });
```

Comment:
We've been putting "version" before "share" everywhere, so why not be consistent here as well?
```suggestion
        Inertia::version(function () use ($request) {
            return $this->version($request);
        });
        
        Inertia::share($this->share($request));
```